src/RealFourierArray.jl

```diff
@@ -0,0 +1,91 @@
+# Lazy structure for an array whose real and fourier part can be accessed by A.real and A.fourier
+# For normalization conventions, see PlaneWaveBasis.jl
+
+import Base.getproperty, Base.setproperty!, Base.fieldnames
+import Base.*, Base.+, Base./
+import Base.eltype
+
+## these functions are minimally optimized for the moment
+
+## TODO this always allocates both real and fourier parts, which might be a bit wasteful
+## TODO implement broadcasting. This seems to be non-trivial, so I just implement a few simple methods
+## TODO implement compressed representation in Fourier space of real arrays
+
+# Tr is the underlying real type
+# T is the type of the array in real space
+# TODO make this concrete by parametrizing on the type of the arrays
+mutable struct RealFourierArray{Tr <: Real, T <: Union{Tr, Complex{Tr}}}
+    basis::PlaneWaveBasis{Tr}
+    _real::AbstractArray{T, 3}
+    _fourier::AbstractArray{Complex{Tr}, 3}
+    _state::Int # 0: only _real set. 1: only _fourier set. 2: both set
+end
+# Type of the real part
+Base.eltype(A::RealFourierArray{Tr, T}) where {Tr, T} = T
```

Comment:
Why is it this, which determines the `eltype`? If we would iterate over a `RealFourierArray` would we expect to get the real-space elements or the fourier-space ones. I'd say the Fourier ones, no? I mean the Fourier ones are the coefficients to our basis, right?

Reply:
Not on the square basis; at the level of RFA there is no preferred representation (at the level of the kpt-dependent basis there is, it's fourier). eltype in fourier is always complex, eltype in real seems more useful (is the quantity real or complex?)

Reply:
Agree. So the RFA is basically a real-space array, which happens to also have a fourier analogue inside. I like that.